modules/claims_api/app/controllers/claims_api/v2/veteran_identifier_controller.rb

```diff
@@ -1,40 +1,34 @@
 # frozen_string_literal: true
 
 require 'evss/error_middleware'
+require 'claims_api/v2/params_validation/veteran_identifier'
 
 module ClaimsApi
   module V2
     class VeteranIdentifierController < ClaimsApi::V2::ApplicationController
-      # TODO: REMOVE BEFORE IMPLEMENTATION
-      skip_before_action :authenticate, only: %i[find]
-      ICN_FOR_TEST_USER = '1012667145V762142'
-      # TODO: REMOVE BEFORE IMPLEMENTATION
+      before_action :validate_params
 
       def find
-        raise ::Common::Exceptions::Unauthorized if request.headers['Authorization'].blank?
-
-        validate_request
         veteran = find_veteran(params)
 
-        render json: { id: veteran[:id] }
+        raise ::Common::Exceptions::ResourceNotFound unless veteran_icn_found?(veteran)
+
+        user_is_the_veteran   = current_user_is_the_veteran?(veteran: veteran, user: @current_user)
+        user_is_a_veteran_rep = current_user_is_a_veteran_representative?(@current_user)
+        raise ::Common::Exceptions::Forbidden unless user_is_the_veteran || user_is_a_veteran_rep
+
+        render json: { id: veteran.mpi.icn }
       end
 
       private
 
-      def validate_request
-        params.require(:ssn)
-        params.require(:birthdate)
-        params.require(:firstName)
-        params.require(:lastName)
-
-        validate_ssn!(params[:ssn])
-        validate_birthdate!(params[:birthdate])
-      end
+      def validate_params
+        validator = ClaimsApi::V2::ParamsValidation::VeteranIdentifier.validator(params)
 
-      def validate_ssn!(ssn)
-        return if ssn.match?(/^\d{9}$/)
+        return if validator.valid?
 
-        raise ::Common::Exceptions::InvalidFieldValue.new('ssn', ssn)
+        validation_error = validator.errors.first
+        raise ::Common::Exceptions::InvalidFieldValue.new(validation_error.attribute, validation_error.type)
```

Comment:
now that we're using this "validation" approach, i wonder if we can actually just pass that "validator" above to this common exceptions class instead:
https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/common/exceptions/validation_errors.rb#L9

Reply:
Ended up having to create a `Common::Exceptions::ValidationErrorsBadRequest` class and using that because we wanted a validation failure to return a `400` error code, rather than the `422` that `Common::Exceptions::ValidationErrors` provides.
That work was done [here](https://github.com/department-of-veterans-affairs/vets-api/pull/7265/commits/f832e1e5c1b967fa8314375bba732d8d5d973911).